traits/trait_list_object.py

```diff
@@ -165,22 +237,22 @@ def __delitem__(self, key):
         ----------
         key : integer or slice
             Index of the element(s) to be deleted.
+
+        Raises
```

Comment:
Drive-by fix unrelated to the main purpose of the PR. This should possibly be pulled out into a separate PR.

Reply:
Done in #1011.